paper-input-container.html

```diff
@@ -266,13 +302,28 @@
     },
 
     _onInput: function(event) {
-      if (event.target.value) {
+      this._handleInput(event.target);
+    },
+
+    _onValueChanged: function(event) {
+      this._handleInput(event.target);
+    },
+
+    _handleInput: function(inputElement) {
+      var value = inputElement[this._propertyForValue] || inputElement.value;
+      if (value) {
         this._inputHasContent = true;
       } else {
         this._inputHasContent = false;
       }
       if (this.autoValidate) {
-        this._inputIsInvalid = !event.target.validity.valid;
+        this._inputIsInvalid = !inputElement.validity.valid;
+      }
+      // notify add-ons
+      var addons = Polymer.dom(this.$.addOnContent).getDistributedNodes();
+      for (var a, i = 0; a = addons[i]; i++) {
+        // this could get called before the distributed node is upgraded
+        a.fire && a.fire('input-value-changed', {inputElement: inputElement, value: value});
```

Comment:
I'm not sure there's an approved way to manually set up a binding outside of the element's template, and that also seems difficult to decipher. Since this is a distributed node an event seem to be the most standard method for communication.